src/Marten/MartenServiceCollectionExtensions.cs

```diff
@@ -0,0 +1,144 @@
+        /// <returns></returns>
+        public static MartenConfigurationExpression AddMarten(this IServiceCollection services, string connectionString)
+        {
+            var options = new StoreOptions();
+            options.Connection(connectionString);
+            return services.AddMarten(options);
+        }
+
+        /// <summary>
+        /// Add Marten IDocumentStore, IDocumentSession, and IQuerySession service registrations
+        /// to your application using the configured StoreOptions
+        /// </summary>
+        /// <param name="services"></param>
+        /// <param name="options">The Marten configuration for this application</param>
+        /// <returns></returns>
+        public static MartenConfigurationExpression AddMarten(this IServiceCollection services, StoreOptions options)
+        {
+            services.AddSingleton(options);
+            services.AddSingleton<IDocumentStore, DocumentStore>();
+
+            // This can be overridden by the expression following
+            services.AddSingleton<ISessionFactory, DefaultSessionFactory>();
+
+
+            services.AddScoped(s => s.GetRequiredService<ISessionFactory>().QuerySession());
+            services.AddScoped(s => s.GetRequiredService<ISessionFactory>().OpenSession());
+
+            return new MartenConfigurationExpression(services, options);
+        }
+
+        /// <summary>
+        /// Add Marten IDocumentStore, IDocumentSession, and IQuerySession service registrations
+        /// to your application using the configured StoreOptions
+        /// </summary>
+        /// <param name="services"></param>
+        /// <param name="configure"></param>
+        /// <returns></returns>
+        public static MartenConfigurationExpression AddMarten(this IServiceCollection services, Action<StoreOptions> configure)
+        {
+            var options = new StoreOptions();
+            configure(options);
+
+            return services.AddMarten(options);
+        }
+
+        public class MartenConfigurationExpression
+        {
+            private readonly IServiceCollection _services;
+            private readonly StoreOptions _options;
+
+            public MartenConfigurationExpression(IServiceCollection services, StoreOptions options)
+            {
+                _services = services;
+                _options = options;
+            }
+
+            /// <summary>
+            /// Use an alternative strategy / configuration for opening IDocumentSession or IQuerySession
+            /// objects in the application with a custom ISessionFactory type registered as a singleton
+            /// </summary>
+            /// <typeparam name="T"></typeparam>
+            /// <returns></returns>
+            public MartenConfigurationExpression BuildSessionsWith<T>() where T : class, ISessionFactory
+            {
+                _services.AddSingleton<ISessionFactory, T>();
```

Comment:
Same comment as above https://github.com/JasperFx/marten/pull/1501/files#r436521253.

Also `ISessionFactory` not always should be Singleton. For a non-default case I'd rather expect it to be Scoped - eg. if the user would like to start the session with Tenant. The tenant probably will be taken from the request.